instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/trace/TraceApiBenchmark.kt

```diff
@@ -0,0 +1,153 @@
+import androidx.benchmark.junit4.measureRepeated
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.datadog.android.Datadog
+import com.datadog.android.DatadogConfig
+import com.datadog.android.sdk.benchmark.aThrowable
+import com.datadog.android.tracing.Tracer
+import com.datadog.tools.unit.invokeMethod
+import fr.xgouchet.elmyr.junit4.ForgeRule
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class TraceApiBenchmark {
+    @get:Rule
+    val benchmark = BenchmarkRule()
+    @get:Rule
+    val forge = ForgeRule()
+
+    lateinit var testedTracer: Tracer
+
+    lateinit var mockWebServer: MockWebServer
+
+    @Before
+    fun setUp() {
+        mockWebServer = MockWebServer()
+            .apply {
+                start()
+            }
+        mockWebServer.setDispatcher(object : Dispatcher() {
+            override fun dispatch(request: RecordedRequest): MockResponse {
+                return mockResponse(200)
+            }
+        })
+        val fakeEndpoint = mockWebServer.url("/").toString().removeSuffix("/")
+
+        val context = InstrumentationRegistry.getInstrumentation().context
+        val config = DatadogConfig
+            .Builder("NO_TOKEN")
+            .useCustomTracesEndpoint(fakeEndpoint)
+            .build()
+        Datadog.initialize(context, config)
+
+        testedTracer = Tracer
+            .Builder()
+            .setPartialFlushThreshold(1)
+            .build()
+    }
+
+    @After
+    fun tearDown() {
+        Datadog.invokeMethod("stop")
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun benchmark_sending_span() {
+        benchmark.measureRepeated {
+            val operationName = runWithTimingDisabled { forge.anAlphabeticalString() }
+            val span = testedTracer.buildSpan(operationName).start()
+            span.finish()
+        }
+    }
+
+    @Test
+    fun benchmark_sending_span_with_throwable() {
+        benchmark.measureRepeated {
+            val (operationName, throwable) = runWithTimingDisabled {
+                forge.anAlphabeticalString() to forge.aThrowable()
+            }
+
+            val span = testedTracer.buildSpan(operationName).start()
+            span.setErrorMeta(throwable)
+            span.finish()
+        }
+    }
+
+    @Test
+    fun benchmark_sending_spans_with_baggage_items_and_logs() {
+        addSpans(1)
+    }
+
+    @Test
+    fun benchmark_sending_medium_load_of_spans() {
+        addSpans(MEDIUM_ITERATIONS)
+    }
+
+    @Test
+    fun benchmark_sending_heavy_load_of_spans() {
+        addSpans(BIG_ITERATIONS)
+    }
+
+    private fun addSpans(iterations: Int) {
```

Comment:
this doesn't add anything, you could rename it `benchmarkCreatingLogsWithTagsAndMatadata`